src/hotspot/share/runtime/threads.cpp

```diff
@@ -799,15 +800,6 @@ jint Threads::create_vm(JavaVMInitArgs* args, bool* canTryAgain) {
     initialize_class(vmSymbols::jdk_internal_vm_Continuation(), CHECK_JNI_ERR);
   }
 
-#if INCLUDE_CDS
-  // capture the module path info from the ModuleEntryTable
-  ClassLoader::initialize_module_path(THREAD);
-  if (HAS_PENDING_EXCEPTION) {
-    java_lang_Throwable::print(PENDING_EXCEPTION, tty);
-    vm_exit_during_initialization("ClassLoader::initialize_module_path() failed unexpectedly");
-  }
-#endif
```

Comment:
Not obvious where this functionality is now handled.

Reply:
It is now being handled in `ClassLoaderDataShared::ensure_module_entry_tables_exist()` and `AOTCodeSourceConfig::dumptime_init_helper()`.

Reply:
How about adding the vm_exit in `ClassLoaderDataShared::ensure_module_entry_table_exist()` instead of assert?

```
void ClassLoaderDataShared::ensure_module_entry_table_exist(oop class_loader) {
  Handle h_loader(JavaThread::current(), class_loader);
  ModuleEntryTable* met = Modules::get_module_entry_table(h_loader);
  if (met == nullptr) {
    vm_exit_during_initialization("ClassLoaderDataShared::ensure_module_entry_table_exist() failed unexpectedly");
  }
}
```

Reply:
I can't answer that. As a refactoring I expect to see the current behaviour preserved.

Reply:
After discussion with Ioi, we found a place to call the vm_exit function:

```
void AOTClassLocationConfig::dumptime_init(JavaThread* current) {
  assert(CDSConfig::is_dumping_archive(), "");
  _dumptime_instance = NEW_C_HEAP_OBJ(AOTClassLocationConfig, mtClassShared);
  _dumptime_instance->dumptime_init_helper(current);
  if (current->has_pending_exception()) {
    // we can get an exception only when we run out of metaspace, but that
    // shouldn't happen this early in bootstrap.
    java_lang_Throwable::print(current->pending_exception(), tty);
    vm_exit_during_initialization("AOTClassLocationConfig::dumptime_init_helper() failed unexpectedly");
  }
}
```